src/validations.ts

```diff
@@ -1,13 +1,39 @@
-import { LogLevel } from './logs';
-import { IPathPosition } from './parsers';
+import { IPathPosition, ObjPath } from './parsers';
+
+export enum ValidationSeverity {
+  Info = 'info',
+  Warn = 'warn',
+  Error = 'error',
+}
+
+export enum ValidationSeverityNum {
+  Info = 30,
+  Warn = 40,
+  Error = 50,
+}
 
 export interface IValidation {
-  ruleId: string;
+  /** Name identifier for this type of validation. */
+  name: string;
+
+  /** Validation severity as a pretty string. */
+  severity: ValidationSeverity;
 
-  level?: LogLevel;
+  /** Validation severity as a number. */
+  severityNum: ValidationSeverityNum;
```

Comment:
The same information is provided in two ways: severity and severityNum. Moreover it could be misprovided by accident, e.g. severity: 'info', severityNum: 40.

Since SeverityNum carries information about the order, why not make it a primary source of severity information? The label could be looked up in types if needed.

Reply:
Great point, I agree. Made numeric severity the default, but left severityLabel as required since I think it is quite useful to have on validations.

https://github.com/stoplightio/types/pull/23/commits/2fa46d692838268c8e662f554a975fe71ddda6d1